handler-proxy/src/main/java/io/netty/handler/proxy/ProxyHandler.java

```diff
@@ -287,7 +287,7 @@ private void setConnectSuccess() {
             connectTimeoutFuture.cancel(false);
         }
 
-        if (connectPromise.trySuccess(ctx.channel())) {
+        if (!connectPromise.isDone()) {
```

Comment:
@Scottmitch nope not expected to be called frequently. So I would prefer to keep it as it is